### PR TITLE
Float layer

### PR DESCRIPTION
**Pull request by nilscb**

### CODE REVIEW

File: react/src/lib/components/DeckGLMap/storybook/DeckGLMap.stories.jsx [hunk 1]
```diff
@@ -421,6 +421,46 @@ Axes.args = {
     },
 };
 
+// Example using "Map" layer. AUses float32 mesh and properties binary arrays. Not PNG.
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/storybook/DeckGLMap.stories.jsx:424]: Can we move this to a `DeckGLMap/layers/map/mapLayer.story.tsx` file?

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/storybook/DeckGLMap.stories.jsx (hunk 1 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/storybook/DeckGLMap.stories.jsx]: Done

File: react/src/lib/components/DeckGLMap/layers/map/terrainMapLayer.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        };
+    }
+
+    decodePickingColor(): number {
+        return 0;
+    }
+
+    // For now, use `any` for the picking types.
+    //eslint-disable-next-line
+    getPickingInfo({ info }: { info: any }): any {
```
[the remark this answers is not in the document]
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/map/terrainMapLayer.ts:220]: maybe try this in a later commit? I was not able to solve it directly.

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts [hunk 3]
```diff
@@ -303,47 +324,108 @@ async function load_mesh_and_texture(
...
-    // Url to png image representing the height mesh.
-    mesh: string;
+    // Url to the height (z values) mesh.
+    meshUrl: string;
 
-    // Bounding box of the terrain mesh, [minX, minY, maxX, maxY] in world coordinates
-    bounds: [number, number, number, number];
+    // Bounding box of the terrain mesh [[origoX, origoY], [nx, ny], [dx, dy]]] in world coordinates.
+    // nx, y: number of cells in each direction. dx, dy: cell width in each direction.
+    bounds: [[number, number], [number, number], [number, number]];
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts:372]: I think we should use the same bounds type and prop name as we decide on in https://github.com/equinor/webviz-subsurface-components/pull/1095

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+};
+const north_arrow_layer = {
+    "@@type": "NorthArrow3DLayer",
+    id: "north-arrow-layer",
+};
+export const MapLayer: ComponentStory<typeof DeckGLMap> = () => {
+    const args = {
+        id: "map",
+        layers: [axes_hugin, meshMapLayer, north_arrow_layer],
+        bounds: [432150, 6475800, 439400, 6481500] as NumberQuad,
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx:89]: Use `frame` instead?

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts [hunk 5]
```diff
@@ -303,47 +327,112 @@ async function load_mesh_and_texture(
...
     meshMaxError: number;
 
-    // Url to png image for map properties. (ex, poro or perm values as a texture)
-    propertyTexture: string;
+    // Url to the properties. (ex, poro or perm values)
+    propertiesUrl: string;
 
-    // Rotates around 'bounds' upper left corner counterclockwise in degrees.
+    // Rotates map counterclockwise in degrees around 'rotPoint' specified below.
     rotDeg: number;
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts:389]: I think we should move `rotDeg` and `rotPoint` to the `frame` prop.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx (hunk 4 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx:89]: Fin this in next pull request?

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+const axes_hugin = {
+    "@@type": "AxesLayer",
+    id: "axes-layer2",
+    bounds: [432150, 6475800, -3500, 439400, 6481500, 0],
+};
+const north_arrow_layer = {
+    "@@type": "NorthArrow3DLayer",
+    id: "north-arrow-layer",
+};
+export const MapLayer: ComponentStory<typeof DeckGLMap> = () => {
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx:85]: I think the bounds need to be adjusted in this story: 

Screenshot from 2022-07-04 07-13-12

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — APPROVED** [review]

**Pull request merged**